makefiles/tools/openocd-adapters/stlink.inc.mk

```diff
@@ -3,15 +3,16 @@
 STLINK_VERSION ?= 2-1
 
 # Use STLINK_VERSION to select which stlink version is used
-OPENOCD_ADAPTER_INIT ?= \
-  -c 'source [find interface/stlink-v$(STLINK_VERSION).cfg]' \
-  -c 'transport select hla_swd'
+OPENOCD_ADAPTER_INIT ?= -c 'source [find interface/stlink-v$(STLINK_VERSION).cfg]'
+
+# ST-Link only supports hla_swd
+OPENOCD_TRANSPORT = hla_swd
```

Comment:
[The documentation](http://openocd.org/doc/html/Debug-Adapter-Configuration.html#SWD-Transport) says the [newer ST-Link (v2 and v3)](http://openocd.org/doc/html/Debug-Adapter-Configuration.html#st_005flink_005fdap_005finterface) support in addition `dapdirect_swd`, which seems to be preferred over `hla_swd` when available.

Reply:
Hm is there some way to detect if this is supported?

I've updated the ST-LINK on my `stm32f429i-disco` to V2J37S0 and OpenOCD gives me

```
Open On-Chip Debugger 0.10.0+dev-01363-g480ba8ca8-dirty (2020-07-27-13:53)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
WARNING: interface/stlink-v2.cfg is deprecated, please switch to interface/stlink.cfg
Error: Debug adapter doesn't support 'dapdirect_swd' transport
```

Reply:
My point wasn't really to handle this, but to update the comment and use `?=` to allow users to use `dapdirect_swd`, if they are using a recent ST-Link with a recent OpenOCD.

```Makefile
# All ST-Link adapters support hla_swd, so use that for simplicity
OPENOCD_TRANSPORT ?= hla_swd
```

Reply:
Alright!
The idea was that a board can set `OPENOCD_TRANSPORT = swd`, so we'll need to prefix that if `OPENOCD_TRANSPORT` is already set.